x-pack/plugins/upgrade_assistant/server/lib/reindexing/reindex_service.ts

```diff
@@ -369,6 +395,48 @@ export const reindexServiceFactory = (
   // ------ The service itself
 
   return {
+    async hasRequiredPrivileges(indexName: string) {
+      // If security is disabled, return true.
+      const xpackInfo = await callCluster('transport.request', {
```

Comment:
I think instead of reimplementing this we should get it from `server.plugins.xpack_main.info` if possible. Then we can use `xpackInfo.feature('security')` to see if it's enabled.

Reply:
Ah awesome, didn't know that existed.